telemetry/definitions/commonDefinitions.json

```diff
@@ -7136,65 +7125,76 @@
             "passive": true
         },
         {
-            "name": "toolkit_initModule",
-            "description": "The Toolkit has completed initialization for the specified module.",
+            "name": "toolkit_invokeAction",
+            "description": "User invoked an action. `source` is the notification that produced the action",
             "metadata": [
                 {
-                    "type": "attempts",
-                    "required": false
+                    "type": "action",
+                    "required": true
                 },
                 {
-                    "type": "duration",
-                    "required": false
+                    "type": "component",
+                    "required": true
                 },
                 {
-                    "type": "module",
+                    "type": "id",
                     "required": true
                 },
+                {
+                    "type": "reason",
+                    "required": false
+                },
                 {
                     "type": "result",
                     "required": true
                 },
                 {
-                    "type": "version",
-                    "required": false
+                    "type": "source",
+                    "required": true
+                }
+            ]
+        },
+        {
+            "name": "toolkit_moduleClose",
```

Comment:
The `namespace-noun-verb` ordering is something I missed in the updated "spec". This is contrary to almost all of our existing metrics, isn't it? `auth_signInPageOpened`, `auth_switchRoles`, `aws_refreshExplorer`.

Can we just define the spec as `namespace-verb-noun`, instead? That will avoid a lot of churn and/or conflicting legacy names.

Reply:
`auth_signInPageOpened` in this case does conform to the `namespace-noun-verb` spec. So I guess this was a bit of an example that we didn't have well defined standards already.

I mentioned the reason for doing`namespace-noun-verb` here: https://github.com/aws/aws-toolkit-common/pull/984/files#diff-809396fc93ef2f298c19a75db76447c555a86af06581d65d65808533cbc10769R33

But I'm open to discussion so we can define some standard for the future

Reply:
Summary of discussion w/ Justin + Chris:

- We will update spec to more closely match our existing metrics
  - `{namespace}_{did|will| }{verb}{noun}`
  - Examples:
    - `toolkit_openModule`
    - `toolkit_willOpenModule`
    - `toolkit_didLoadModule`

- IF we didn't have these legacy metrics we would have considered a different scheme
  - Example: `{namespace}_{noun}_{did|will| }{verb}`
  - But to reduce confusion since all our existing metrics tend to follow {verb}{noun}